bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/CTabFolderRenderer.java

```diff
@@ -318,9 +318,14 @@ protected Point computeSize (int part, int state, GC gc, int wHint, int hHint) {
 					Image image = item.getImage();
 					if (image != null && !image.isDisposed()) {
 						Rectangle bounds = image.getBounds();
-						if ((state & SWT.SELECTED) != 0 || parent.showUnselectedImage) {
-							width += bounds.width;
+						//If image is not visible, tab width should not include image width
```

Comment:
Sure. We'll re-format it. 
Also, we are working on extracting this logic into a separate method  `getTextPadding(...) ` for uniform padding calculation when image is not visible. This will simplify the code.